process/sync/storageBootstrap/shardStorageBootstrapper.go

```diff
@@ -139,6 +140,94 @@ func (ssb *shardStorageBootstrapper) cleanupNotarizedStorage(shardHeaderHash []b
 	}
 }
 
+func (ssb *shardStorageBootstrapper) cleanupNotarizedStorageForHigherNoncesIfExist(
+	crossNotarizedHeaders []bootstrapStorage.BootstrapHeaderInfo,
+) {
+	var numConsecutiveNoncesNotFound int
+
+	lastCrossNotarizedNonce, err := getLastCrossNotarizedHeaderNonce(crossNotarizedHeaders)
+	if err != nil {
+		log.Warn("cleanupNotarizedStorageForHigherNoncesIfExist", "error", err.Error())
+		return
+	}
+
+	log.Debug("cleanup notarized storage has been started", "from nonce", lastCrossNotarizedNonce+1)
+	nonce := lastCrossNotarizedNonce
+
+	for {
+		nonce++
+
+		metaBlock, metaBlockHash, err := process.GetMetaHeaderFromStorageWithNonce(
+			nonce,
+			ssb.store,
+			ssb.uint64Converter,
+			ssb.marshalizer,
+		)
+		if err != nil {
+			log.Debug("meta block is not found in MetaHdrNonceHashDataUnit storage",
+				"nonce", nonce, "error", err.Error())
+
+			numConsecutiveNoncesNotFound++
+			if numConsecutiveNoncesNotFound > maxNumOfConsecutiveNoncesNotFoundAccepted {
+				log.Debug("cleanup notarized storage has been finished",
+					"from nonce", lastCrossNotarizedNonce+1,
+					"to nonce", nonce)
+				break
+			}
+
+			continue
+		}
+
+		numConsecutiveNoncesNotFound = 0
+		nonceToByteSlice := ssb.uint64Converter.ToByteSlice(metaBlock.GetNonce())
+		err = ssb.store.GetStorer(dataRetriever.MetaHdrNonceHashDataUnit).Remove(nonceToByteSlice)
+		if err != nil {
+			log.Debug("meta block was not removed from MetaHdrNonceHashDataUnit storage",
+				"shardId", metaBlock.GetShardID(),
+				"nonce", metaBlock.GetNonce(),
+				"hash", metaBlockHash,
+				"error", err.Error())
+		} else {
+			log.Debug("meta block has been removed from MetaHdrNonceHashDataUnit storage",
+				"shardId", metaBlock.GetShardID(),
+				"nonce", metaBlock.GetNonce(),
+				"hash", metaBlockHash)
+		}
+
+		err = ssb.store.GetStorer(dataRetriever.MetaBlockUnit).Remove(metaBlockHash)
```

Comment:
can you extract a method to removeMetaFromMetaBlockUnit

Reply:
Done